SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+    _log "  -h, --help                  Print this help page"
+    _log "  --accpy VERSION             Version of Acc-Py to be used (options: ${ACCPY_ALL_VERSIONS_STR})"
+}
+
+# --------------------------------------------------------------------------------------------
+
+# Parse command line arguments
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --env|-e)
+            ENV_NAME=$2
+            shift
+            shift
+            ;;
+        --req|-r)
+            REQUIREMENTS=$2
+            shift
+            shift
+            ;;
+        --clear|-c)
+            CLEAR_ENV=--clear
+            shift
+            ;;
+        --help|-h)
+            print_help
+            exit 0
+            ;;
+        --accpy)
+            ACCPY_CUSTOM_VERSION=$2
+            shift
+            shift
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+ENV_PATH="/home/$USER/${ENV_NAME}"
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if an environment with the same name was already created, if --clear is not passed
+if [ -d "$ENV_PATH" ] && [ -z "$CLEAR_ENV" ]; then
+    >&2 _log "ERROR: Virtual environment already exists."
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_CUSTOM_VERSION" ] && [ ! -e "$ACCPY_BASE/$ACCPY_CUSTOM_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version. Options: ${ACCPY_ALL_VERSIONS_STR}"
+    exit 1
+fi
+
+# Checks if a requirements file is given
+if [ -z "$REQUIREMENTS" ]; then
+    >&2 _log "ERROR: No requirements provided." && _log
+    print_help
+    exit 1
+# Checks if the provided requirements source is found
+elif [ -f $REQUIREMENTS ]; then
+    if [[ ${REQUIREMENTS##*.} != "txt" ]]; then
+        >&2 _log "ERROR: Invalid requirements file."
+        exit 1
+    fi
+    REQ_PATH=$REQUIREMENTS
+elif [[ $REQUIREMENTS == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REQUIREMENTS)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+    REPO_PATH=$HOME/SWAN_projects/${repo_name}
+
+    # Checks if the repository already exists in the home directory
+    if [ -d "${REPO_PATH}" ]; then
+        if [ -z "$CLEAR_ENV" ]; then
+            >&2 _log "ERROR: ${REPO_PATH} already exists. Use --clear to reclone the repository."
+            exit 1
+        else
+            rm -rf ${REPO_PATH}
+        fi
+    fi
+
+    echo "Cloning the repository from ${REQUIREMENTS}..."
+    # Clone the repository
+    git clone $REQUIREMENTS -q --template /usr/share/git-core/templates "${REPO_PATH}" || { >&2 _log "ERROR: Failed to clone repository"; exit 1; }
```

Comment:
Why do we need this `--template` option?

Reply:
The only reason I used it was to fix an error I was getting before I add that flag.

Reply:
Maybe due to the fact it was a closed environment. When I "reopen" it, I'll change that too, to see if everything's fine without it.

Reply:
What do you mean closed environment? What happens if you remove it now and let the extension run and call this script?